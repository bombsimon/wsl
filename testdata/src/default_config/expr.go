package testpkg

import "fmt"

func fn() {
	a := 1
	b := 2
	c := 3
	a = 4
	b = 3
}

func fn2() {
	a := 1
	b := 2
	fmt.Println("") // want "missing whitespace decreases readability"
	c := 3          // want "missing whitespace decreases readability"
	d := 4
}

func fn3() {
	a := 1
	b := 2
	fmt.Println(b)
	c := 3 // want "missing whitespace decreases readability"
	d := 4
}
