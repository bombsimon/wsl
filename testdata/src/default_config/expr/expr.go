package testpkg

import "fmt"

func fn() {
	a := 1
	b := 2
	c := 3
	a = 4
	b = 3

	_ = a
	_ = b
	_ = c
}

func fn2() {
	a := 1
	b := 2
	fmt.Println("") // want `missing whitespace above this line \(no shared variables above expr\)`
	c := 3          // want `missing whitespace above this line \(invalid statement above assign\)`
	d := 4

	_ = a
	_ = b
	_ = c
	_ = d
}

func fn3() {
	a := 1
	b := 2
	fmt.Println(b)
	c := 3 // want `missing whitespace above this line \(invalid statement above assign\)`
	d := 4

	_ = a
	_ = b
	_ = c
	_ = d
}

func fn4() {
	b := 2
	<-ch // want `missing whitespace above this line \(no shared variables above expr\)`
}

func fn5() {
	s := []string{
		func() string { // want +1 `unnecessary whitespace \(leading-whitespace\)`

			"a"
		},
		"b",
	}
}

func issue153(b bool) {
	for i := 0; i < 4; i++ {
		fmt.Println()
	}
	Up(3) // want `missing whitespace above this line \(invalid statement above expr\)`

	if autoheight != 3 {
		fmt.Printf("height should be 3 but is %d", autoheight)
	}
	Down(3) // want `missing whitespace above this line \(invalid statement above expr\)`

	if b {
		fmt.Println()
	}
	Up(b) // want `missing whitespace above this line \(invalid statement above expr\)`
}
