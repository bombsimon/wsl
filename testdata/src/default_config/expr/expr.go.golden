package testpkg

import "fmt"

func fn() {
	a := 1
	b := 2
	c := 3
	a = 4
	b = 3

	_, _, _ = a, b, c
}

func fn2() {
	a := 1
	b := 2

	fmt.Println("") // want `missing whitespace above this line \(no shared variables above expr\)`

	c := 3          // want `missing whitespace above this line \(invalid statement above assign\)`
	d := 4

	_, _, _, _ = a, b, c, d
}

func fn3() {
	a := 1
	b := 2
	fmt.Println(b)

	c := 3 // want `missing whitespace above this line \(invalid statement above assign\)`
	d := 4

	_, _, _, _ = a, b, c, d
}

func fn4(ch chan struct{}) {
	b := 2

	<-ch // want `missing whitespace above this line \(no shared variables above expr\)`

	_, _ = b, ch
}

func fn5() {
	s := []string{
		func() string { // want +1 `unnecessary whitespace \(leading-whitespace\)`
			return "a"
		}(),
		"b",
	}

	_ = s
}

func Up(_ bool)  {}
func Down(_ int) {}

func issue153(autoheight int, b bool) {
	for i := 0; i < 4; i++ {
		fmt.Println()
	}

	Up(true) // want `missing whitespace above this line \(invalid statement above expr\)`

	if autoheight != 3 {
		fmt.Printf("height should be 3 but is %d", autoheight)
	}

	Down(3) // want `missing whitespace above this line \(invalid statement above expr\)`

	if b {
		fmt.Println()
	}

	Up(b) // want `missing whitespace above this line \(invalid statement above expr\)`
}

